feat: allow pg_search to not be in shared_preload_libraries on PG 17 (#2148)
paradedb/paradedb: docs/deploy/self-hosted/extensions.mdx

## Update `postgresql.conf`

> Next, add the extension(s) to `shared_preload_libraries` in `postgresql.conf`. Adding `pg_search` to `shared_preload_libraries` is unnecessary if your Postgres version 17 or higher.

Reviewer: ```suggestion
Next, add the extension(s) to `shared_preload_libraries` in `postgresql.conf`. Adding `pg_search` to `shared_preload_libraries` is unnecessary if your Postgres version is 17 or higher.
```

Reviewer: Sorry, I made a typo myself

> Next, add the extension(s) to `shared_preload_libraries` in `postgresql.conf`. Adding `pg_search` to `shared_preload_libraries` is unnecessary if your Postgres version is less than 17.

Reviewer: ```suggestion
Next, add the extension(s) to `shared_preload_libraries` in `postgresql.conf`. Adding `pg_search` to `shared_preload_libraries` is unnecessary if your Postgres version 17 or higher.
```

Reviewer: Assuming that's what you meant?
